03_letters/marchitect/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func letters(s string) map[rune]int {
+	m := make(map[rune]int)
+	// keep a list of what characters exist in the string s
+	chars := ""
+	for _, c := range s {
+		if !strings.ContainsAny(chars, string(c)) {
```

Comment:
This is inefficient. ContainsAny is `O(n)` and so it `strings.Count(s, string(c))`, updating a map is `O(1)`.
I accept that runtime is irrelevant for the size of strings expected in this program, but it is good practise to be aware. 
```go
for _, c := range s {
    m[c]++
}
```
should do the trick